examples/tv-casting-app/linux/main.cpp

```diff
@@ -185,6 +195,71 @@ void InitCommissioningFlow(intptr_t commandArg)
+}
+
+void OnContentLauncherFailureResponse(void * context, uint8_t status)
+{
+    ChipLogProgress(AppServer, "ContentLauncher: Default Failure Response: 0x%02x", status);
+}
+
+void DeviceEventCallback(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    if (event->Type == DeviceLayer::DeviceEventType::kCommissioningComplete)
+    {
+        Server * server = &(chip::Server::GetInstance());
+
+        chip::NodeId tvNodeId             = chip::DeviceLayer::Internal::DeviceControlServer::DeviceControlSvr().GetPeerNodeId();
+        chip::FabricIndex peerFabricIndex = chip::DeviceLayer::Internal::DeviceControlServer::DeviceControlSvr().GetFabricIndex();
+
+        chip::FabricInfo * fabric = server->GetFabricTable().FindFabricWithIndex(peerFabricIndex);
+        if (fabric == nullptr)
+        {
+            ChipLogError(AppServer, "Did not find fabric for index %d", peerFabricIndex);
+            return;
+        }
+
+        chip::DeviceProxyInitParams initParams = {
+            .sessionManager = &(server->GetSecureSessionManager()),
+            .exchangeMgr    = &(server->GetExchangeManager()),
+            .idAllocator    = &(server->GetSessionIDAllocator()),
+            .fabricInfo     = fabric,
+            // TODO: Determine where this should be instantiated
+            .imDelegate = chip::Platform::New<chip::Controller::DeviceControllerInteractionModelDelegate>(),
+        };
+
+        PeerId peerID = fabric->GetPeerId();
+        peerID.SetNodeId(tvNodeId);
+
+        chip::OperationalDeviceProxy * operationalDeviceProxy =
+            chip::Platform::New<chip::OperationalDeviceProxy>(initParams, peerID);
+        if (operationalDeviceProxy == nullptr)
+        {
+            ChipLogError(AppServer, "Failed in creating an instance of OperationalDeviceProxy");
+            return;
+        }
+
+        SessionHandle handle = server->GetSecureSessionManager().FindSecureSessionForNode(tvNodeId);
+        operationalDeviceProxy->SetConnectedSession(handle);
+
+        chip::Controller::ContentLauncherCluster cluster;
+        CHIP_ERROR err = cluster.Associate(operationalDeviceProxy, kTvEndpoint);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer, "Associate() failed: %" CHIP_ERROR_FORMAT, err.Format());
+            return;
+        }
+        chip::CharSpan contentURL(kContentUrl, strlen(kContentUrl)), displayString(kContentDisplayStr, strlen(kContentDisplayStr));
+        chip::Callback::Callback<DefaultSuccessCallback> * onSuccessCallback =
+            new chip::Callback::Callback<DefaultSuccessCallback>(OnContentLauncherSuccessResponse, 0);
+        chip::Callback::Callback<DefaultFailureCallback> * onFailureCallback =
+            new chip::Callback::Callback<DefaultFailureCallback>(OnContentLauncherFailureResponse, 0);
+        cluster.LaunchURL(onSuccessCallback->Cancel(), onFailureCallback->Cancel(), contentURL, displayString);
```

Comment:
Please don't use the old Callback-based cluster APIs: we are trying to remove the last few consumers.  See the `InvokeCommand` API in `src/controller/CHIPCluster.h` instead (which you can use on `cluster` here).